File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left, right = 0, 0
        res = 0
        window_vals = set()

        while right < len(s):
            if s[right] in window_vals:
                window_vals.remove(s[left])
                left += 1 
            else:
                window_vals.add(s[right])
                res = max(res, right - left + 1)
                right += 1

        return res 

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let result = 0
        let left = 0
        let right = 0

        while (right < prices.length) {
            let profit = prices[right] - prices[left]

            if(prices[left] > prices[right]){
                left = right
            }else{
                result = Math.max(profit, result)
            }

            right++
        }



        return result
    }
}
